oauth2_http/java/com/google/auth/oauth2/OAuth2Credentials.java

```diff
@@ -94,6 +95,11 @@ public void refresh() throws IOException {
       requestMetadata = null;
       temporaryAccess = null;
       temporaryAccess = refreshAccessToken();
+      if (changeListeners != null) {
+        for (CredentialsChangedListener listener : changeListeners) {
```

Comment:
Doh. I missed that synchronized. This could still have problems with concurrent modification, but that seems unlikely given there isn't a `removeChangeListener`